### PR TITLE
ui/networking: show "connecting" immediately after clicking on a network.

### PR DESCRIPTION
**Pull request by deanlee**
`emit refreshSignal()` when connecting to network to fix this old bug.

### CODE REVIEW

File: selfdrive/ui/qt/offroad/networking.h [hunk 1]
```diff
@@ -12,8 +12,9 @@ class WifiItem : public QWidget {
   void setItem(const Network& n, const QPixmap &icon, bool show_forget_btn, const QPixmap &strength);
 
 signals:
-  void connectToNetwork(const Network &n);
-  void forgotNetwork(const Network &n);
+  // Cannot pass Network by reference. it may change after the signal is sent.
```
Comment (deanlee) [inline selfdrive/ui/qt/offroad/networking.h:15]: Also fixes a bug caused by passing network by references. (introduced in  #29037)

Comment (sshane) [reply selfdrive/ui/qt/offroad/networking.h:15]: Can you make that fix a new PR?

Comment (deanlee) [reply selfdrive/ui/qt/offroad/networking.h:15]: split to https://github.com/commaai/openpilot/pull/29448

**Comment by sshane** [comment]
does this ever happen? perhaps it was the bug fixed in https://github.com/commaai/openpilot/pull/29396 causing it not to show as connecting for a few seconds https://github.com/commaai/openpilot/pull/29396

**Comment by deanlee** [comment]
think this still happen. although the connected state can be refreshed correctly after #29396, The UI still won't be refreshed before the refresh timer is activated.

**Comment by sshane** [comment]
I think things should be better after https://github.com/commaai/openpilot/pull/29396, I believe we get a refresh signal as soon as network manager starts connecting, so it's fine to wait until then. I checked on my Pixel and there also is a delay before it shows that it's connecting.

**Comment by adeebshihadeh** [comment]
This is a real improvement.

**Comment by sshane** [comment]
we can merge this before https://github.com/commaai/openpilot/pull/29647, since it adds the infrastructure for the other PR

**Pull request merged**